cmake doesn't look for filteraudio headers

cmake can't find filteraudio headers when filteraudio is not system-wide installed, which is the case when, for example, cross-compiling qTox to Windows. You can reproduce my issue by cross-compiling qTox to Windows with https://github.com/qTox/qTox/pull/4588 changes.

filteraudio doesn't provide pkg-config files, so cmake [falls back to using `find_library`](https://github.com/qTox/qTox/blob/5d159e0203bdd0e72153cce9ca5c0fe8483ba0db/cmake/Dependencies.cmake#L62). It does find the library successfully, but the issue is that it only looks for the library, when it also should be looking for the headers in the addition. This is what I get

```
-- FILTERAUDIO LIBRARY_DIRS: 
-- FILTERAUDIO INCLUDE_DIRS: 
-- FILTERAUDIO CFLAGS_OTHER: 
-- FILTERAUDIO LIBRARIES:    /workspace/i686/dep-cache/libfilteraudio/lib/libfilteraudio.a
-- FILTERAUDIO found
-- using filteraudio
```

As you can see, `INCLUDE_DIRS` is empty. And then I get

```
In file included from /build/qtox/src/audio/audio.cpp:23:0:
/build/qtox/src/audio/backend/openal2.h:41:26: fatal error: filter_audio.h: No such file or directory
compilation terminated.
```

To fix this, [besides just doing `find_library()` call]((https://github.com/qTox/qTox/blob/5d159e0203bdd0e72153cce9ca5c0fe8483ba0db/cmake/Dependencies.cmake#L62)), it should be doing something like

```
find_path(${pkg}_INCLUDE_DIRS NAMES filter_audio.h)
```

You'd want to either pass `filter_audio.h` as an extra argument to `search_dependency()`, or write a [`FindFilteraudio.cmake` module](https://cmake.org/Wiki/CMake:How_To_Find_Libraries) and use `find_package(filteraudio)`.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -538,7 +538,7 @@ if(${ENABLE_GTK_SYSTRAY})
 endif()
 
 if(${USE_FILTERAUDIO})
-    search_dependency(FILTERAUDIO LIBRARY filteraudio OPTIONAL)
+    search_dependency(FILTERAUDIO LIBRARY filteraudio HEADER filter_audio.h OPTIONAL)
     if(${FILTERAUDIO_FOUND})
         set(${PROJECT_NAME}_SOURCES ${${PROJECT_NAME}_SOURCES}
             src/audio/backend/openal2.cpp

--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -35,7 +35,7 @@ include(Qt5CorePatches)
 
 function(search_dependency pkg)
   set(options OPTIONAL)
-  set(oneValueArgs PACKAGE LIBRARY FRAMEWORK)
+  set(oneValueArgs PACKAGE LIBRARY FRAMEWORK HEADER)
   set(multiValueArgs)
   cmake_parse_arguments(arg "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
 
@@ -60,7 +60,8 @@ function(search_dependency pkg)
   # Last, search for the library itself globally.
   if(NOT ${pkg}_FOUND AND arg_LIBRARY)
     find_library(${pkg}_LIBRARIES NAMES ${arg_LIBRARY})
-    if(${pkg}_LIBRARIES)
+    find_path(${pkg}_INCLUDE_DIRS NAMES ${arg_HEADER})
+    if(${pkg}_LIBRARIES AND ${pkg}_INCLUDE_DIRS)
       set(${pkg}_FOUND TRUE)
     endif()
   endif()

